File: RandytheStudent V2.0/RandyTheStudent/RandyTheStudent/BasicTable.m
//
//  BasicTable.m
//  Magma
//
//  Created by Enrique Galicia on 19/08/13.
//  Copyright (c) 2013 Enrique Galicia. All rights reserved.
//

#import "BasicTable.h"

@interface BasicTable ()

@end

@implementation BasicTable
@synthesize delegadobase=_delegadobase,tamsubtit,tamtit,funcion;


- (id)initWithStyle:(UITableViewStyle)style
{
    self = [super initWithStyle:style];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    cantidaddeelementos=0;
    valoresphp=[[NSMutableDictionary alloc]init];

}
-(void)cargartablas:(NSDictionary*)informacion{

    titulo=[informacion objectForKey:@"Titulo"];
    titulo2=[titulo copy];
    subtitulo=[informacion objectForKey:@"Subtitulo"];
    subtitulo2=[subtitulo copy];


    cantidaddeelementos=[titulo2 count];
    [self.tableView reloadData];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - Table view data source


- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    if (cantidaddeelementos>0) {
        return cantidaddeelementos;
    }
    else{
    return 1;
    }

}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *CellIdentifier = @"CountryCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    if (cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:CellIdentifier];
    }
    
	NSString *tituloss = [titulo2 objectAtIndex:indexPath.row];
	cell.textLabel.text = tituloss;
    cell.textLabel.lineBreakMode = NSLineBreakByCharWrapping;
    cell.textLabel.numberOfLines = 1;
    cell.textLabel.font = [UIFont  fontWithName:@"Futura-CondensedMedium" size:tamtit];
    cell.backgroundColor=[UIColor whiteColor];
    
    NSString *subtituloss =  [subtitulo objectAtIndex:indexPath.row];
    cell.detailTextLabel.text=subtituloss;
    cell.detailTextLabel.numberOfLines = 2;
    cell.detailTextLabel.font = [UIFont fontWithName:@"Futura-CondensedMedium" size:tamsubtit];
    return cell;
}


- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView cellForRowAtIndexPath:indexPath];
    NSRange Alto = [cell.textLabel.text rangeOfString:@"_"];
    if (Alto.location!=2147483647) {
        NSString *ID=[cell.textLabel.text substringWithRange:NSMakeRange(0, Alto.location)];
        NSString *TareaAsignar=ID;
        [self.delegadobase setselected:TareaAsignar fun:funcion];
    }
    else{
        [self.delegadobase setselected:cell.textLabel.text fun:funcion];
    }

    
    
    
}


@end

File: RandytheStudent V2.0/RandyTheStudent/RandyTheStudent/BasicTable.swift
//
//  BasicTable.swift
//  RandyTheStudent
//
//  Swift port of the original BasicTable.m (Objective-C) — a small
//  reusable UITableViewController used as an embedded subview (its
//  `.view` gets manually re-parented; see Classes.m). Loads BasicTable.xib
//  via the standard "nib name matches class name" convention, which is
//  why `@objc(BasicTable)` matters here, not just for ObjC interop:
//  NSStringFromClass(self) has to come back as "BasicTable" for that nib
//  lookup to find the file.
//
//  `@objc(basictable)` on the delegate protocol keeps Classes.h's (still
//  Objective-C, until it's converted) `<basictable, ComboDelegate>`
//  conformance declaration compiling unchanged.
//
//  One deliberate change: `delegadobase` was `retain` (strong) in the
//  original, which is a real retain cycle (Classes owns TAlumnos/TGrupos/
//  TActividades strongly, and each held Classes strongly right back via
//  this delegate). Made it `weak`, standard Cocoa delegate practice and
//  harmless here since Classes always outlives these child controllers.
//

import UIKit

@objc(basictable)
protocol BasicTableDelegate: NSObjectProtocol {
    @objc(setselected:fun:)
    func setselected(_ seleccion: String, fun: Int32)
}

@objc(BasicTable)
class BasicTable: UITableViewController {

    @objc weak var delegadobase: BasicTableDelegate?
    @objc var tamtit: Int32 = 0
    @objc var tamsubtit: Int32 = 0
    @objc var funcion: Int32 = 0
    @objc var informacion: NSDictionary?

    private var titulo: [Any] = []
    private var titulo2: [Any] = []
    private var subtitulo: [Any] = []
    private var subtitulo2: [Any] = []
    private var cantidaddeelementos = 0

    override func viewDidLoad() {
        super.viewDidLoad()
        cantidaddeelementos = 0
    }

    @objc(cargartablas:)
    func cargartablas(_ informacion: NSDictionary) {
        titulo = (informacion["Titulo"] as? [Any]) ?? []
        titulo2 = titulo
        subtitulo = (informacion["Subtitulo"] as? [Any]) ?? []
        subtitulo2 = subtitulo

        cantidaddeelementos = titulo2.count
        tableView.reloadData()
    }

    // MARK: - Table view data source

    override func numberOfSections(in tableView: UITableView) -> Int {
        1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        cantidaddeelementos > 0 ? cantidaddeelementos : 1
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cellIdentifier = "CountryCell"
        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier)
            ?? UITableViewCell(style: .subtitle, reuseIdentifier: cellIdentifier)

        let tituloss = indexPath.row < titulo2.count ? "\(titulo2[indexPath.row])" : ""
        cell.textLabel?.text = tituloss
        cell.textLabel?.lineBreakMode = .byCharWrapping
        cell.textLabel?.numberOfLines = 1
        cell.textLabel?.font = UIFont(name: "Futura-CondensedMedium", size: CGFloat(tamtit))
        cell.backgroundColor = .white

        let subtituloss = indexPath.row < subtitulo.count ? "\(subtitulo[indexPath.row])" : ""
        cell.detailTextLabel?.text = subtituloss
        cell.detailTextLabel?.numberOfLines = 2
        cell.detailTextLabel?.font = UIFont(name: "Futura-CondensedMedium", size: CGFloat(tamsubtit))
        return cell
    }

    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        guard let cell = tableView.cellForRow(at: indexPath) else { return }
        let text = cell.textLabel?.text ?? ""
        if let range = text.range(of: "_") {
            let id = String(text[text.startIndex..<range.lowerBound])
            delegadobase?.setselected(id, fun: funcion)
        } else {
            delegadobase?.setselected(text, fun: funcion)
        }
    }
}
